R/httr2-translate.R

```diff
@@ -0,0 +1,189 @@
+#' Translate httr2 request to curl command
+#'
+#' Convert an httr2 request object to equivalent curl command line syntax.
+#' This is useful for debugging, sharing requests, or converting to other tools.
+#'
+#' @param .req An httr2 request object created with [request()].
+#' @return A character string containing the curl command.
+#' @export
+#' @examples
+#' \dontrun{
+#' # Basic GET request
+#' request("https://httpbin.org/get") |>
+#'   httr2_translate()
+#'
+#' # POST with JSON body
+#' request("https://httpbin.org/post") |>
+#'   req_body_json(list(name = "value")) |>
+#'   httr2_translate()
+#'
+#' # POST with form data
+#' request("https://httpbin.org/post") |>
+#'   req_body_form(name = "value") |>
+#'   httr2_translate()
+#' }
+httr2_translate <- function(.req) {
+  # validate the request
+  check_request(.req)
+
+  # Extract URL
+  url <- .req$url
+
+  # use the request's method if it is set, otherwise infer
+  method <- .req$method %||%
+    {
+      if (!is.null(.req$body$data)) {
+        "POST"
+      } else {
+        "GET"
+      }
+    }
+
+  # we will append to cmd_parts to build up the request
+  cmd_parts <- c("curl")
+
+  # if the method isn't GET, it needs to be specified with `-X`
+  if (method != "GET") {
+    cmd_parts <- c(cmd_parts, paste0("-X ", method))
+  }
+
+  # if headers are present, add them using -H flag
+  if (!is.null(.req$headers) && length(.req$headers) > 0) {
+    headers <- .req$headers
+    for (name in names(headers)) {
+      value <- headers[[name]]
+
+      # handle weakrefs
+      if (rlang::is_weakref(value)) {
+        value <- rlang::wref_value(value)
+      }
+
+      # unobfuscate obfuscated
+      if (is_obfuscated(value)) {
+        value <- unobfuscate(value, handle = "reveal")
+      }
+
+      cmd_parts <- c(cmd_parts, paste0('-H "', name, ': ', value, '"'))
+    }
+  }
+
+  # manage options
+  if (!is.null(.req$options) && length(.req$options) > 0) {
+    options <- .req$options
+    for (name in names(options)) {
```

Comment:
I think you need a comment here explaining the rationale for why these options were mapped. I don't see any obvious pattern.